saleor/graphql/warehouse/mutations.py

```diff
@@ -59,6 +61,62 @@ def prepare_address(cls, cleaned_data, *args):
         return address_form.save()
 
 
+class WarehouseShippingZoneAssign(WarehouseMixin, ModelMutation, I18nMixin):
+    warehouse = graphene.Field(
+        Warehouse, description="A warehouse to add shipping zone."
+    )
+
+    class Meta:
+        model = models.Warehouse
+        permissions = (ProductPermissions.MANAGE_PRODUCTS,)
+        description = "Add shipping zone to given warehouse."
+        error_type_class = WarehouseError
+        error_type_field = "warehouse_errors"
+
+    class Arguments:
+        id = graphene.ID(description="ID of a warehouse to update.", required=True)
+        shipping_zone_id = graphene.ID(
+            description="ID of a shipping zone to assign.", required=True
+        )
+
+    @classmethod
+    def perform_mutation(cls, _root, info, **data):
+        warehouse = cls.get_node_or_error(info, data.get("id"), only_type=Warehouse)
+        shipping_zone = cls.get_node_or_error(
+            info, data.get("shipping_zone_id"), only_type=ShippingZone
+        )
+        warehouse.shipping_zones.add(shipping_zone)
+        return WarehouseShippingZoneAssign(warehouse=warehouse)
+
+
+class WarehouseShippingZoneUnassign(WarehouseMixin, ModelMutation, I18nMixin):
+    warehouse = graphene.Field(
+        Warehouse, description="A warehouse to add shipping zone."
+    )
+
+    class Meta:
+        model = models.Warehouse
+        permissions = (ProductPermissions.MANAGE_PRODUCTS,)
+        description = "Remove shipping zone from given warehouse."
+        error_type_class = WarehouseError
+        error_type_field = "warehouse_errors"
+
+    class Arguments:
+        id = graphene.ID(description="ID of a warehouse to update.", required=True)
+        shipping_zone_id = graphene.ID(
```

Comment:
This could also be a list of IDs to unassign multiple at once. Same for the mutation above to assign shipping zones.